lib/features/artifacts.js

```diff
@@ -128,7 +128,22 @@ export default class Artifacts {
       };
 
       try {
-        await uploadArtifact(taskHandler, stream, entryName, expiry, headers);
+        await new Promise((accept, reject) => {
+          let numRetries = taskHandler.runtime.retry.numRetries;
+          let retry = () => {
+            uploadArtifact(taskHandler, stream, entryName, expiry, headers)
```

Comment:
Ok, so I spoke with Jonas and he left some comments below in the tests.  It's true that reading from the stream should cause the data to be drained from it, so subsequent reads should not work as expected because it's not the data you think it is.  

The reason why the test is passing is explained below (blocking 443 causes createArtifact to fail before it even reads from the stream)